include/jwt-cpp/jwt.h

```diff
@@ -2668,6 +2668,162 @@ namespace jwt {
+		 * \throw std::bad_cast Claim was present but not a string (Should not happen in a valid token)
+		 */
+		typename json_traits::string_type get_key_id() const { return get_jwks_claim("kid").as_string(); }
+
+		/**
+		 * Get x5c claim
+		 * \return x5c as string
+		 * \throw std::runtime_error If claim was not present
+		 * \throw std::bad_cast Claim was present but not a string (Should not happen in a valid token)
+		 */
+		typename json_traits::string_type get_x5c() const { return json_traits::as_string(get_jwks_claim("x5c").as_array().front()); } //do not use serialize() instead
+
+		/**
+		 * Check if algortihm is present ("alg")
+		 * \return true if present, false otherwise
+		 */
+		bool has_algorithm() const noexcept { return has_jwks_claim("alg"); }
+
+		/**
+		 * Check if key id is present ("kid")
+		 * \return true if present, false otherwise
+		 */
+		bool has_key_id() const noexcept { return has_jwks_claim("kid"); }
+
+		/**
+		 * Check if x5c is present ("x5c")
+		 * \return true if present, false otherwise
+		 */
+		bool has_x5c_id() const noexcept { return has_jwks_claim("x5c"); }
+
+		/**
+		 * Check if a jwks claim is present
+		 * \return true if claim was present, false otherwise
+		 */
+		bool has_jwks_claim(const typename json_traits::string_type& name) const noexcept { return jwks_claims.count(name) != 0; }
+
+		/**
+		 * Get jwks claim
+		 * \return Requested claim
+		 * \throw std::runtime_error If claim was not present
+		 */
+		basic_claim_t get_jwks_claim(const typename json_traits::string_type& name) const {
+			if (!has_jwks_claim(name)) {
+				typename json_traits::string_type error_msg = "claim \"";
+				error_msg += name;
+				error_msg += "\" not found";
+				throw std::runtime_error(error_msg.c_str());
+			}
+			return jwks_claims.at(name);
+		}
+
+		bool empty() const noexcept { return jwks_claims.empty(); }
+
+	protected:
+		JWT_CLAIM_EXPLICIT jwks(const typename json_traits::value_type& claim) {
+			for (const auto& e : json_traits::as_object(claim)) {
+				jwks_claims.emplace(e.first, basic_claim_t(e.second));
+			}
+		}
+
+	private:
+		std::unordered_map<typename json_traits::string_type, basic_claim_t> jwks_claims;
+	};
+
+	template<typename json_traits>
+	class jwks_keys {
+		using basic_claim_t = basic_claim<json_traits>;
+		using jwks_t = jwks<json_traits>;
+
+	public:
+
+		JWT_CLAIM_EXPLICIT jwks_keys(const typename json_traits::string_type& jwks_keys_token) {
+			auto parse_claims = [](const typename json_traits::string_type& str) {
+				std::unordered_map<typename json_traits::string_type, jwks_t> res;
+				typename json_traits::value_type val;
+				if (!json_traits::parse(val, str))
+					throw std::runtime_error("Invalid json");
+
+				for (const auto& k : json_traits::as_object(val)) {
+					for (const typename json_traits::value_type& item : json_traits::as_array(k.second)) {
+						jwks_t jwks_entry(item);
+						res.emplace(jwks_entry.get_key_id(), std::move(jwks_entry));
```

Comment:
What if there is no KID in a JWK? there's still a lot of auth server which do not follow this best practice